src/apps/testapps/testVec3d.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <float.h>
+#include <math.h>
+#include <stdlib.h>
+#include "test.h"
+#include "vec3d.h"
+
+BEGIN_TESTS(Vec3d);
+
+TEST(_pointSquareDist) {
+    Vec3d v1 = {0, 0, 0};
+    Vec3d v2 = {1, 0, 0};
+    Vec3d v3 = {0, 1, 1};
+    Vec3d v4 = {1, 1, 1};
+    Vec3d v5 = {1, 1, 2};
+
+    t_assert(fabs(_pointSquareDist(&v1, &v1)) < DBL_EPSILON,
+             "distance to self is 0");
+    t_assert(fabs(_pointSquareDist(&v1, &v2) - 1) < DBL_EPSILON,
+             "distance to <1,0,0> is 1");
+    t_assert(fabs(_pointSquareDist(&v1, &v3) - 2) < DBL_EPSILON,
+             "distance to <0,1,1> is 2");
+    t_assert(fabs(_pointSquareDist(&v1, &v4) - 3) < DBL_EPSILON,
+             "distance to <1,1,1> is 3");
+    t_assert(fabs(_pointSquareDist(&v1, &v5) - 6) < DBL_EPSILON,
+             "distance to <1,1,2> is 6");
+}
+
+TEST(_geoToVec3d) {
+    Vec3d origin = {0};
```

Comment:
For my info - why not `{0, 0, 0}`? Doesn't this result in undefined memory for the uninitialized struct fields? Or is this functionally equivalent to `{0, 0, 0}`?

Reply:
I understand this should be the same as initializing all members to 0, https://stackoverflow.com/questions/11152160/initializing-a-struct-to-0